### PR TITLE
Add (IL)linker opt-in flag

### PR DESCRIPTION
**Pull request by ViktorHofer**
This is similar to the compiler package which overrides what's inbox in dotnet/sdk. Exposing the illinker via a property as an opt-in to guarantee that the right package is used throughout the repository as msbuild caches tasks and even with ALC there is currently no way to prevent that.

TODO post merge:
- Add a subscription from mono/linker --> arcade
- Remove subscription in dotnet/runtime and enable the opt-in flag

### CODE REVIEW

**Comment by marek-safar** [comment]
> Remove subscription in dotnet/runtime and enable the opt-in flag

What would be the benefit of it? The reason for the subscription is to have fast updates in runtime which with arcade won't be possible

**Comment by ViktorHofer** [comment]
> What would be the benefit of it? The reason for the subscription is to have fast updates in runtime which with arcade won't be possible

Same as for dotnet/roslyn (in regards to their compiler package), that would make sure that core repositories in the stack use a consistent version of the linker so that sourcebuild doesn't need to build multiple different versions.

> to have fast updates in runtime which with arcade won't be possible

Arcade ingests roughly 3 times per week and we are fine with that pacing for the compiler package. If we need faster updates for ILLink, we can keep the subscription in dotnet/runtime.

**Comment by marek-safar** [comment]
> If we need faster updates for ILLink, we can keep the subscription in dotnet/runtime.

I believe we do.

File: src/Microsoft.DotNet.Arcade.Sdk/tools/DefaultVersions.props [hunk 1]
```diff
@@ -55,6 +55,12 @@
       Repo can set this property to true if it needs to use a different version of the compiler than the one in the dotnet SDK.
     -->
     <UsingToolMicrosoftNetCompilers Condition="'$(UsingToolMicrosoftNetCompilers)' == ''">false</UsingToolMicrosoftNetCompilers>
+
+    <!--
+      Use linkers from the Microsoft.NET.ILLink.Tasks package.
+      Repo can set this property to true if it needs to use a different version of the linker than the one in the dotnet SDK.
+    -->
+    <UsingToolMicrosoftNetLinkers Condition="'$(UsingToolMicrosoftNetLinkers)' == ''">false</UsingToolMicrosoftNetLinkers>
```
Comment (akoeplinger) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/DefaultVersions.props:63]: I'd prefer if we match the property name with the package name like we do for `UsingToolMicrosoftNetCompilers`:

```suggestion
      Use IL linker from the Microsoft.NET.ILLink.Tasks package.
      Repo can set this property to true if it needs to use a different version of the IL linker than the one in the dotnet SDK.
    -->
    <UsingToolMicrosoftNetILLinkTasks Condition="'$(UsingToolMicrosoftNetILLinkTasks)' == ''">false</UsingToolMicrosoftNetILLinkTasks>
```

File: src/Microsoft.DotNet.Arcade.Sdk/tools/Settings.props [hunk 2]
```diff
@@ -21,6 +21,7 @@
   <Import Project="Workarounds.props"/>
 
   <Import Project="Compiler.props" Condition="'$(UsingToolMicrosoftNetCompilers)' == 'true'" />
+  <Import Project="Linker.props" Condition="'$(UsingToolMicrosoftNetLinkers)' == 'true'" />
```
Comment (akoeplinger) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/Settings.props:24]:
```suggestion
  <Import Project="Linker.props" Condition="'$(UsingToolMicrosoftNetILLinkTasks)' == 'true'" />
```

**Review by akoeplinger — APPROVED** [review]

**Pull request merged**